Lib/test/test_functools.py

```diff
@@ -2478,6 +2478,17 @@ def test_access_from_class(self):
     def test_doc(self):
         self.assertEqual(CachedCostItem.cost.__doc__, "The cost of the item.")
 
+    def test_isabstractmethod(self):
+        class AbstractExpensiveCalculator(abc.ABC):
+            @functools.cached_property
+            @abc.abstractmethod
+            def calculate(self):
+                pass
+
+        self.assertTrue(getattr(AbstractExpensiveCalculator.calculate, '__isabstractmethod__', False))
+        with self.assertRaises(TypeError):
+            AbstractExpensiveCalculator()
+
```

Comment:
Can you add a test on a method which doesn't have the ``__isabstractmethod__`` attribute? I expect such test:

```
self.assertFalse(hasattr(AbstractExpensiveCalculator.calculate, '__isabstractmethod__'))
```